synapse/api/constants.py

```diff
@@ -57,6 +57,7 @@ class JoinRules:
 
 class LoginType:
     PASSWORD = "m.login.password"
+    APPSERVICE = "uk.half-shot.msc2778.login.application_service"
```

Comment:
Is this the right place for this? LoginType looks like user-interactive auth stuff, while the /login types are specified at the start of `class LoginRestServlet`.

Reply:
good point, will update.